src/zerg/CombatZerg.ts

```diff
@@ -150,7 +150,7 @@ export class CombatZerg extends Zerg {
 	 */
 	autoMelee(possibleTargets = this.room.hostiles) {
 		const target = CombatTargeting.findBestCreepTargetInRange(this, 1, possibleTargets)
-					   || CombatTargeting.findBestStructureTargetInRange(this, 1);
+					   || CombatTargeting.findBestStructureTargetInRange(this, 1,false); //disabled allowUnowned structure attack in order not to desrtory poison walls
```

Comment:
this shouldn't modify the base combat zerg methods -- perhaps add a flag to room memory which says don't destroy poison walls?